lib/page/needed.rb

```diff
@@ -0,0 +1,31 @@
+module Page
+  class Needed
+    def initialize
+      make_client
```

Comment:
`make_client` has two branches in it, but every client you're returning here seems to only follow one of those paths.